java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java

```diff
@@ -2956,4 +2959,80 @@ void testContiguousSplitConstructor() {
+      HostColumnVector.ColumnBuilder.StructData retData2 = hcv1.getStruct(1, type);
+      HostColumnVector.ColumnBuilder.StructData retData3 = hcv1.getStruct(2, type);
+      HostColumnVector.ColumnBuilder.StructData retData4 = hcv1.getStruct(3, type);
+      assertEquals(data1, retData1.dataRecord);
+      assertEquals(data2, retData2.dataRecord);
+      assertEquals(data3, retData3.dataRecord);
+      assertEquals(data4, retData4);
+      assertStructColumnsAreEqual(expected, columnVector, type);
+    }
+  }
+
+  @Test
+  void testStructChildValidity() {
+    HostColumnVector.ColumnBuilder.StructType type = new HostColumnVector.ColumnBuilder.StructType(true, 4);
+    type.addChild(new HostColumnVector.ColumnBuilder.BasicType(true, 4, DType.INT32));
+    type.addChild(new HostColumnVector.ColumnBuilder.BasicType(true, 4, DType.INT64));
+    List data1 = Arrays.asList(1, 2L);
+    List data2 = Arrays.asList(4, 5L);
+    List data3 = null;
+    List data4 = Arrays.asList(8, null);
+    HostColumnVector.ColumnBuilder.StructData structData1 = new HostColumnVector.ColumnBuilder.StructData(data1);
+    HostColumnVector.ColumnBuilder.StructData structData2 = new HostColumnVector.ColumnBuilder.StructData(data2);
+    HostColumnVector.ColumnBuilder.StructData structData3 = new HostColumnVector.ColumnBuilder.StructData(data3);
+    HostColumnVector.ColumnBuilder.StructData structData4 = new HostColumnVector.ColumnBuilder.StructData(data4);
+    try (HostColumnVector hcv = HostColumnVector.fromStructs(type, Arrays.asList(structData1, structData2, structData3, structData4));
+         ColumnVector columnVector = hcv.copyToDevice();
+         HostColumnVector hcv1 = columnVector.copyToHost();
+         ColumnVector expected = hcv1.copyToDevice()) {
+      assertFalse(hcv.isNull(0));
+      assertFalse(hcv.isNull(1));
+      assertTrue(hcv.isNull(2));
+      assertFalse(hcv.isNull(3));
+      HostColumnVector.NestedHostColumnVector intChildCol = hcv.children.get(0);
+      HostColumnVector.NestedHostColumnVector longChildCol = hcv.children.get(1);
+      assertFalse(intChildCol.isNull(0));
+      assertFalse(intChildCol.isNull(1));
+      assertTrue(intChildCol.isNull(2));
+      assertFalse(intChildCol.isNull(3));
+      assertFalse(longChildCol.isNull(0));
+      assertFalse(longChildCol.isNull(1));
+      assertTrue(longChildCol.isNull(2));
+      assertTrue(longChildCol.isNull(3));
+
+      intChildCol = hcv1.children.get(0);
+      longChildCol = hcv1.children.get(1);
+
+      assertFalse(intChildCol.isNull(0));
+      assertFalse(intChildCol.isNull(1));
+      assertTrue(intChildCol.isNull(2));
+      assertFalse(intChildCol.isNull(3));
+      assertFalse(longChildCol.isNull(0));
+      assertFalse(longChildCol.isNull(1));
+      assertTrue(longChildCol.isNull(2));
+      assertTrue(longChildCol.isNull(3));
+      assertStructColumnsAreEqual(expected, columnVector, type);
+    }
```

Comment:
@revans2 , this is a test for children hostCV validity vector, and the code change that allows this behavior is 
[here](https://github.com/rapidsai/cudf/pull/6171/files#diff-7515abd63357ad5368785ee947cb0823R1347), appreciate any comments on the correctness, I do think it is doing what column factory on cpp side does, but I dont know why just making a column_view of type struct doesnt do the magic for us without having to do this.

Reply:
I thought the magic happened when you created a struct column, not a column_view.  Or else creating a column_view is going to be very expensive, which it is not supposed to be.

Reply:
Agree, that seems to be the case, I just wanted to make sure I understood this correctly.

Reply:
@revans2 , I think I have covered most of the concerns, but happy to get ur reviews on any remaining lacunae.